parlai/core/agents.py

```diff
@@ -131,30 +155,44 @@ def __init__(self, opt, shared=None):
 
     # return state/action dict based upon passed state
     def act(self):
+        """Act upon the previous observation."""
         if self.observation is not None and 'text' in self.observation:
             t = {'text': 'Hello agent!'}
         return t
 
     def epoch_done(self):
+        """Return whether the epoch is done."""
         return self.epochDone
 
     # Default unknown length
     def num_examples(self):
+        """
+        Return the number of examples (individual utterances) in the dataset.
```

Comment:
Should we maybe say `... number of examples (e.g. individual utterances)...` to specify that "individual utterances" is more of an example/archetype than the absolute norm for each dataset? E.g. for captioning datasets this will be an image, not an utterance.